packages/twenty-server/src/modules/messaging/message-import-manager/drivers/imap/services/imap-find-sent-folder.service.ts

```diff
@@ -0,0 +1,167 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { isNumber } from 'class-validator';
+import { ListResponse, type ImapFlow } from 'imapflow';
+
+/**
+ * Service to find sent folder using IMAP special-use flags
+ *
+ * This service uses IMAP special-use extension (RFC 6154) to identify
+ * the sent folder by looking for the \Sent flag rather than relying on
+ * folder names which can vary across providers and locales.
+ *
+ * Falls back to regex-based detection if special-use flags are not available.
+ * The regex pattern is inspired by imapsync's comprehensive folder mapping.
+ */
+@Injectable()
+export class ImapFindSentFolderService {
+  private readonly logger = new Logger(ImapFindSentFolderService.name);
+
+  public async findSentFolder(client: ImapFlow): Promise<string | null> {
+    try {
+      const list = await client.list();
+
+      this.logger.debug(
+        `Available folders: ${list.map((item) => item.path).join(', ')}`,
+      );
+
+      const specialUseSentFolder = await this.findSentFolderBySpecialUse(
+        client,
+        list,
+      );
+
+      if (specialUseSentFolder) {
+        return specialUseSentFolder;
+      }
+
+      const candidateSentFolder = await this.findSentFolderByCandidates(
+        client,
+        list,
+      );
+
+      if (candidateSentFolder) {
+        return candidateSentFolder;
+      }
+
+      this.logger.warn(
+        'No sent folder found. Only inbox messages will be imported.',
+      );
+
+      return null;
+    } catch (error) {
+      this.logger.warn(`Error listing folders: ${error.message}`);
+
+      return null;
+    }
+  }
+
+  private async findSentFolderBySpecialUse(
+    client: ImapFlow,
+    list: ListResponse[],
+  ): Promise<string | null> {
+    for (const folder of list) {
+      if (folder.specialUse && folder.specialUse.includes('\\Sent')) {
+        this.logger.log(
+          `Found sent folder via special-use flag: ${folder.path}`,
+        );
+
+        const hasMessages = await this.validateFolderHasMessages(
+          client,
+          folder.path,
+        );
+
+        if (hasMessages) {
+          return folder.path;
+        }
+
+        this.logger.warn(
+          `Special-use sent folder "${folder.path}" is empty, checking other candidates`,
+        );
+
+        break;
+      }
+    }
+
+    return null;
+  }
+
+  private async findSentFolderByCandidates(
```

Comment:
let's change the whole "candidate" by regex? or at least candidateByRegex everywhere